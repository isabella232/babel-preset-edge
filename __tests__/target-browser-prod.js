import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Target: Browser: Prod", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    target: "browser",
    env: "production"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
