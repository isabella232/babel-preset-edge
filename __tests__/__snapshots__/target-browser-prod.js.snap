// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Target: Browser: Prod comments 1`] = `"import _camelCase from\\"lodash/camelCase\\";function translate(a){return _camelCase(a)}"`;

exports[`Target: Browser: Prod deadcode 1`] = `"var STATIC=!1;!1,before(),!1,after();"`;

exports[`Target: Browser: Prod emotion 1`] = `"import _jsx from\\"@babel/runtime/helpers/esm/jsx\\";import{css as _css}from\\"emotion\\";var className=/*#__PURE__*/_css({name:\\"1vdp66k-className\\",styles:\\"color:hotpink;label:className;\\"}),_ref=/*#__PURE__*/_jsx(\\"div\\",{className:className},void 0,\\"Some hotpink text.\\");function SomeComponent(){return _ref}"`;

exports[`Target: Browser: Prod es2015: classes 1`] = `"import _inheritsLoose from\\"@babel/runtime/helpers/esm/inheritsLoose\\";export var Component=/*#__PURE__*/function(a){function b(){var b;return b=a.call(this)||this,b.flag=!0,b}_inheritsLoose(b,a);var c=b.prototype;return c.main=function main(){alert(this.foo)},b}(Other);"`;

exports[`Target: Browser: Prod es2015: for_off 1`] = `"import\\"core-js/modules/es7.symbol.async-iterator\\";import\\"core-js/modules/es6.symbol\\";import\\"core-js/modules/web.dom.iterable\\";function test(){for(var a=tasks,b=Array.isArray(a),c=0,a=b?a:a[Symbol.iterator]();;){var d;if(b){if(c>=a.length)break;d=a[c++]}else{if(c=a.next(),c.done)break;d=c.value}try{executeCommands()}catch(a){console.error(\\"Error\\")}}}"`;

exports[`Target: Browser: Prod es2015: generator 1`] = `"function*idMaker(){for(var a=0;;)yield a++}"`;

exports[`Target: Browser: Prod es2015: set 1`] = `"import\\"core-js/modules/web.dom.iterable\\";import\\"core-js/modules/es6.array.iterator\\";import\\"core-js/modules/es6.object.to-string\\";import\\"core-js/modules/es6.string.iterator\\";import\\"core-js/modules/es6.set\\";new Set([1,2,3]);"`;

exports[`Target: Browser: Prod es2016: exponentiation 1`] = `"var result=Math.pow(2,3);"`;

exports[`Target: Browser: Prod esnext: async 1`] = `"function wait(){return new Promise(function(a,b){return Promise.resolve(loadData()).then(function(){try{return a()}catch(a){return b(a)}},b)})}"`;

exports[`Target: Browser: Prod esnext: async_loop 1`] = `"import\\"core-js/modules/es7.symbol.async-iterator\\";import\\"core-js/modules/es6.symbol\\";import\\"core-js/modules/web.dom.iterable\\";function executeTasks(a){return new Promise(function(b,c){function d(){if(g){if(h>=f.length)return[1];i=f[h++]}else{if(h=f.next(),h.done)return[1];i=h.value}var a=function(){try{return d}catch(a){return c(a)}},b=function(){try{return console.error(\\"Error\\"),a()}catch(a){return c(a)}};try{return Promise.resolve(executeCommands()).then(function(){try{return a()}catch(a){return b(a)}},b)}catch(a){b(a)}}function e(){return b()}var f,g,h,i;f=a,g=Array.isArray(f),h=0,f=g?f:f[Symbol.iterator]();var j;return(j=function(a){for(;a;){if(a.then)return void a.then(j,c);try{if(a.pop){if(a.length)return a.pop()?e.call(this):a;a=d}else a=a.call(this)}catch(a){return c(a)}}}.bind(this))(d)})}"`;

exports[`Target: Browser: Prod esnext: catch_bind 1`] = `"try{throw 0}catch(a){doSomethingWhichDoesntCareAboutTheValueThrown()}finally{doSomeCleanup()}"`;

exports[`Target: Browser: Prod esnext: class properties 1`] = `"export var Component=function(){this.onClick=function(){}};"`;

exports[`Target: Browser: Prod esnext: decorators 1`] = `"var _class,MyClass=annotation(_class=function(){})||_class;function annotation(a){a.annotated=!0}"`;

exports[`Target: Browser: Prod esnext: dynamic import 1`] = `"import(/*webpackChunkName:'lodash-wBYk8'*/\\"lodash\\").then(function(a){console.log(a.camelCase(\\"hello world\\"))});"`;

exports[`Target: Browser: Prod esnext: object spread 1`] = `"import\\"core-js/modules/es6.object.assign\\";var original={bar:10},variant=Object.assign({foo:1},original);"`;

exports[`Target: Browser: Prod loadable 1`] = `"import loadable from\\"loadable-components\\";export var Home=loadable({chunkName:function chunkName(){return\\"Home-JfPYC\\"},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"Home-JfPYC\\" */\\"./views/Home\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/Home\\"):require(\\"path\\").resolve(__dirname,\\"./views/Home\\")}});export var About=loadable({chunkName:function chunkName(){return\\"About-8yDxR\\"},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"About-8yDxR\\" */\\"./views/About\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/About\\"):require(\\"path\\").resolve(__dirname,\\"./views/About\\")}});export var Contact=loadable({chunkName:function chunkName(){return\\"Contact-AIf6m\\"},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"Contact-AIf6m\\" */\\"./views/Contact\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/Contact\\"):require(\\"path\\").resolve(__dirname,\\"./views/Contact\\")}});"`;

exports[`Target: Browser: Prod loadable magic-comments 1`] = `"import loadable from\\"loadable-components\\";export var Home=loadable({chunkName:function chunkName(){return\\"Home-JfPYC\\"},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"Home-JfPYC\\" */\\"./views/Home\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?eval(\\"module.require\\")(id):__webpack_require__(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/Home\\"):require(\\"path\\").resolve(__dirname,\\"./views/Home\\")}});export var About=loadable({chunkName:function chunkName(){return\\"About\\"},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"About\\" */\\"./views/About\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/About\\"):require(\\"path\\").resolve(__dirname,\\"./views/About\\")}});export var Contact=loadable({chunkName:function chunkName(){return\\"Contact-AIf6m\\"},isReady:function isReady(a){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */\\"./views/Contact\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/Contact\\"):require(\\"path\\").resolve(__dirname,\\"./views/Contact\\")}});"`;

exports[`Target: Browser: Prod loadable wildcard 1`] = `"import\\"core-js/modules/es6.regexp.replace\\";import loadable from\\"loadable-components\\";var page=\\"About\\",lang=\\"de_DE\\";export var views=loadable({chunkName:function chunkName(){return\\"views-About\\".replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g,\\"-\\")},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"views-[request]\\" */\\"./views/About\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./views/About\\"):require(\\"path\\").resolve(__dirname,\\"./views/About\\")}});export var leanIntl=loadable({chunkName:function chunkName(){return(\\"lean-intl-\\"+lang+\\"-data\\").replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g,\\"-\\")},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"lean-intl-[request]-data\\" */\\"./lean-intl/\\"+lang+\\"/data\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./lean-intl/\\"+lang+\\"/data\\"):require(\\"path\\").resolve(__dirname,\\"./lean-intl/\\"+lang+\\"/data\\")}});export var reactIntl=loadable({chunkName:function chunkName(){return(\\"react-intl-\\"+lang+\\"-data\\").replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g,\\"-\\")},isReady:function isReady(){return\\"undefined\\"!=typeof __webpack_modules__&&!!__webpack_modules__[this.resolve(a)]},requireAsync:function requireAsync(){return import(/* webpackChunkName: \\"react-intl-[request]-data\\" */\\"./react-intl/\\"+lang+\\"/data\\")},requireSync:function requireSync(props){var id=this.resolve(props);return\\"undefined\\"==typeof __webpack_require__?__webpack_require__(id):eval(\\"module.require\\")(id)},resolve:function resolve(){return require.resolveWeak?require.resolveWeak(\\"./react-intl/\\"+lang+\\"/data\\"):require(\\"path\\").resolve(__dirname,\\"./react-intl/\\"+lang+\\"/data\\")}});"`;

exports[`Target: Browser: Prod lodash: async 1`] = `"import _parallel from\\"async/parallel\\";_parallel([]);"`;

exports[`Target: Browser: Prod lodash: lodash 1`] = `"import _camelCase from\\"lodash/camelCase\\";_camelCase(\\"hello world\\");"`;

exports[`Target: Browser: Prod lodash: recompose 1`] = `"import _compose from\\"recompose/compose\\";_compose([]);"`;

exports[`Target: Browser: Prod macro: gql 1`] = `"var query={kind:\\"Document\\",definitions:[{kind:\\"OperationDefinition\\",operation:\\"query\\",name:{kind:\\"Name\\",value:\\"User\\"},variableDefinitions:[],directives:[],selectionSet:{kind:\\"SelectionSet\\",selections:[{kind:\\"Field\\",name:{kind:\\"Name\\",value:\\"user\\"},arguments:[{kind:\\"Argument\\",name:{kind:\\"Name\\",value:\\"id\\"},value:{kind:\\"IntValue\\",value:\\"5\\"}}],directives:[],selectionSet:{kind:\\"SelectionSet\\",selections:[{kind:\\"Field\\",name:{kind:\\"Name\\",value:\\"lastName\\"},arguments:[],directives:[]},{kind:\\"FragmentSpread\\",name:{kind:\\"Name\\",value:\\"UserEntry1\\"},directives:[]}]}}]}}],loc:{start:0,end:79,source:{body:\\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",name:\\"GraphQL request\\",locationOffset:{line:1,column:1}}}};"`;

exports[`Target: Browser: Prod macro: preval 1`] = `"var one=1;"`;

exports[`Target: Browser: Prod promise 1`] = `"import\\"core-js/modules/es6.promise\\";import\\"core-js/modules/es6.object.to-string\\";export default function(){return new Promise(function(a){return a(42)})}"`;

exports[`Target: Browser: Prod react: jsx 1`] = `"import _jsx from\\"@babel/runtime/helpers/esm/jsx\\";var _ref=/*#__PURE__*/_jsx(\\"h1\\",{},void 0,\\"Hello\\");function render(){return _ref}"`;

exports[`Target: Browser: Prod react: jsx spread 1`] = `"import\\"core-js/modules/es6.object.assign\\";function render(a){return React.createElement(\\"h1\\",Object.assign({className:\\"large\\"},a),\\"Hello\\")}"`;

exports[`Target: Browser: Prod react: proptypes 1`] = `"import _jsx from\\"@babel/runtime/helpers/esm/jsx\\";import _inheritsLoose from\\"@babel/runtime/helpers/esm/inheritsLoose\\";import React from\\"react\\";var MyComponent=/*#__PURE__*/function(a){function b(){return a.apply(this,arguments)||this}_inheritsLoose(b,a);var c=b.prototype;return c.render=function render(){return _jsx(\\"div\\",{},void 0,this.props.userName)},b}(React.Component);"`;

exports[`Target: Browser: Prod require context 1`] = `"\\"test\\"===process.env.NODE_ENV&&require(\\"babel-plugin-require-context-hook/register\\")();var loader=require.context(__dirname,!1,/\\\\.js/);"`;
